context.go

```diff
@@ -44,20 +44,15 @@ func (c Context) Array(key string, arr LogArrayMarshaler) Context {
 		c.l.context = arr.write(c.l.context)
 		return c
 	}
-	var a *Array
-	if aa, ok := arr.(*Array); ok {
-		a = aa
-	} else {
-		a = Arr()
-		arr.MarshalZerologArray(a)
-	}
+	a := Arr()
+	arr.MarshalZerologArray(a)
 	c.l.context = a.write(c.l.context)
 	return c
 }
 
 // Object marshals an object that implement the LogObjectMarshaler interface.
 func (c Context) Object(key string, obj LogObjectMarshaler) Context {
-	e := newEvent(LevelWriterAdapter{io.Discard}, 0)
+	e := c.l.scratchEvent()
```

Comment:
Centralized the creation of "scratch" `Event` and moved to the `Logger`. 

We need to make sure we set the event's copies of the `context.Context` (in `.ctx`, ), stack-dump flag (in .stack), and the hooks (in `.hooks` for `Logger`, in `.ch` for `Event`)

All instances use `io.Discard` and 0 == `zerolog.DebugLevel`.